cps-global-lib/src/main/java/org/jenkinsci/plugins/workflow/cps/global/GlobalRepoHookListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jenkinsci.plugins.workflow.cps.global;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import javax.inject.Inject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+public abstract class GlobalRepoHookListener implements ExtensionPoint {
+
+    public abstract void hook();
+
+    @Extension
+    @Restricted({NoExternalUse.class})
+    public static class UserDefinedGlobalVariableRepoListener extends GlobalRepoHookListener {
```

Comment:
:ant: put this inside `UserDefinedGlobalVariableList` where it logically belongs.